lib/matplotlib/artist.py

```diff
@@ -93,6 +100,46 @@ class Artist:
 
     zorder = 0
 
+    def __init_subclass__(cls):
+        # Inject custom set() methods into the subclass with signature and
+        # docstring based on the subclasses' properties.
+        # docstring based on the subclasses' properties.
+
+        if 'set' in cls.__dict__:
+            return  # don't overwrite set if the subclass defines set itself.
+
+        def set(self, **kwargs):
+            Artist.set(self, **kwargs)
+
+        cls.set = set
+        set.__doc__ = ("Set properties.\n\nSupported properties are:\n\n"
+                       f"%({cls.__name__}:kwdoc)s")
+        from matplotlib import docstring
+        docstring.dedent_interpd(set)
+        cls._update_set_signature()
+
+    _PROPERTIES_EXCLUDED_FROM_SET = [
+        'navigate_mode',  # not a user-facing function
+        'figure',         # changing the figure is such a profound operation
+                          # that we don't want this in set()
+        '3d_properties',  # cannot be used as a keyword do to leading digit
```

Comment:
This can come in via a bit of trickery:

```python
obj.set(**{'3d_properties' : ...})
```

but it is probably OK if we do not document that...

Reply:
I also think `set_3d_properties` should not be public API: https://github.com/matplotlib/matplotlib/pull/19572#pullrequestreview-600200619. So let's leave it out.

Reply:
typo: "due to"